### DISCUSSION TITLE
Hello, can I create some msys env with different package version set?

### DISCUSSION THREAD

**Discussion by stirwl** [msys2/MINGW-packages#21490] [Q&A]
I have some legacy code, which can only be built in old packages.
Now I have to switch package versions manually to build old and new software.
Is there a easy way to keep different msys2 enviroment at the same time?

**Comment by Biswa96** [comment]
> Is there a easy way to keep different msys2 enviroment at the same time?

There is no easy way. You have to either install the old package from mirror https://mirror.msys2.org/msys/x86_64/ or build it. We could try to help with any issues related to msys2 and updating legacy code.

**Comment by ghost** [comment]
May you can install MSYS2 twice with a different directory and shortcuts name.
One of them is keep the old packages, the other you can feel free to use new packages you want.
Biswa96 answered me it is permissible to install MSYS2 twice. See https://github.com/msys2/MSYS2-packages/discussions/4846.

**Comment by Kreijstal** [comment]
maybe recreate a repo using the versions you need
